gradlew

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env sh
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+
+##############################################################################
+##
+##  Gradle start up script for UN*X
```

Comment:
<!--new_thread; commit:a4b23fed0e2990e9589a2a69f36008e8626a1eaf; resolved:0-->
Why do we have this or need it?

Reply:
Its the recommended way for users to use the build without needing to install the version of gradle that this build expects (4.2.1). Its also the recommended approach when using Jenkins. Gradle will automatically download its own version and execute itself via the gradlew script. Ditto for the `.bat` file for windows users.

For developers who only develop on this project, they can install Gradle themselves but having this script removes the need for them to do so.